src/App.jsx

```diff
@@ -14,33 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'Sort alphabetically';
+const SORT_FIELD_BY_LENGTH = 'Sort by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (sortField.sortType) {
+    case SORT_FIELD_ALPHABETICALLY:
+      return preparedGoods.sort((good1, good2) => good1.localeCompare(good2));
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    case SORT_FIELD_BY_LENGTH:
+      return preparedGoods.sort((good1, good2) => good1.length - good2.length);
+
+    default:
+      return preparedGoods;
+  }
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState({
+    sortType: '',
+    isReversed: false,
+  });
+
+  let sortedGoods = getPreparedGoods(goodsFromServer, sortField);
+
+  function reverseGoods(goods) {
+    return [...goods].reverse();
+  }
+
+  if (sortField.isReversed === true) {
+    sortedGoods = reverseGoods(sortedGoods);
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() =>
+            setSortField({
+              sortType: SORT_FIELD_ALPHABETICALLY,
+              isReversed: sortField.isReversed,
+            })
+          }
+          className={cn('button', 'is-info', {
+            'is-light': sortField.sortType !== SORT_FIELD_ALPHABETICALLY,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField.sortType !== SORT_FIELD_BY_LENGTH,
+          })}
+          onClick={() =>
+            setSortField({
+              sortType: SORT_FIELD_BY_LENGTH,
+              isReversed: sortField.isReversed,
+            })
+          }
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': sortField.isReversed !== true,
```

Comment:
```suggestion
            'is-light': !sortField.isReversed
```
You don't need to compare with the boolean variable